packages/@ama-sdk/schematics/schematics/code-generator/open-api-cli-generator/open-api-cli.generator.ts

```diff
@@ -79,12 +80,12 @@ export class OpenApiCliGenerator extends CodeGenerator<OpenApiCliOptions> {
 
   /** @inheritDoc */
   protected runCodeGeneratorFactory = (factoryOptions: { rootDirectory?: string } = {}) => {
-    const rootDirectory = factoryOptions.rootDirectory ?
-      (path.isAbsolute(factoryOptions.rootDirectory) ? factoryOptions.rootDirectory : path.resolve(process.cwd(), factoryOptions.rootDirectory)) :
-      process.cwd();
+    const rootDirectory = factoryOptions.rootDirectory
+      ? (path.isAbsolute(factoryOptions.rootDirectory) ? factoryOptions.rootDirectory : path.resolve(process.cwd(), factoryOptions.rootDirectory))
+      : process.cwd();
     return async (generatorOptions?: OpenApiCliOptions) => {
       if (!generatorOptions) {
-        return Promise.reject('Missing options to run open api generator');
+        throw 'Missing options to run open api generator';
```

Comment:
Not sure for this rule (which may be in conflict with the enforcing of `new Error()` when throwing

Reply:
rule `unicorn/no-useless-promise-resolve-reject`